modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Enrich.scala

```diff
@@ -168,42 +180,80 @@ object Enrich {
     BadRow.GenericError(processor, failure, rawPayload)
   }
 
-  def sinkChunk[F[_]: Concurrent: Parallel, B](
+  def sinkChunk[F[_]: Concurrent: Parallel, A, B](
     chunk: List[(B, Result)],
-    sinkOne: Validated[BadRow, EnrichedEvent] => F[Unit],
-    trackLatency: Option[Long] => F[Unit]
-  ): F[List[B]] =
-    chunk.parTraverse {
-      case (record, result) =>
-        val (events, collectorTstamp) = result
-        events.parTraverse_(one => sinkOne(one)).as(record) <* trackLatency(collectorTstamp)
-    }
+    env: Environment[F, A]
```

Comment:
I'm only pointing this out in case you find it interesting:

You never use the type parameter `B`, which means you could drop B from the method signature like this:

```
 def sinkChunk[F[_]: Concurrent: Parallel, A](
    chunk: List[(Any, Result)],
    env: Environment[F, A]
): F[Unit] = {
```

It works because List and Tuple2 are covariant in B.